File: r/day01_part2.R

fname <- "../day1.txt"

input <- as.integer(readLines(fname))

freq_seq <- cumsum(input)

freq_dup <- duplicated(freq_seq)
while (!any(freq_dup)) {
  freq_seq <- c(freq_seq, tail(freq_seq, 1) + freq_seq)
  freq_dup <- duplicated(freq_seq)
}

print(freq_seq[freq_dup][1])

File: python/day01_part2.py
import sys

try:
	fname = sys.argv[1]
except:
	fname = '../day1.txt'

with open(fname, 'r') as f:
	input = [int(n.strip()) for n in f.readlines()]


i = 0
freq = 0
freqs_seen = set()

while freq not in freqs_seen:
	
	freqs_seen.add(freq)
	freq += input[i]
	
	i = (i + 1) % len(input)


print(freq)
